src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
-        where TSerializer : IMemoryMessageBodySerializer, new()
+        where TSerializer : IMemoryMessageBodySerializer
     {
+        private readonly IMemoryMessageBodySerializer serializer;
```

Comment:
I kept it because that type will serialize itself correctly now - if using our serializer